cijoe/tests/linkandload/test_xnvme_consumption.py

```diff
@@ -0,0 +1,103 @@
+"""
+This tests whether:
+
+* It is possible to load the shared library
+* It is possible to link with the shared library
+* It is possible to link with the static library
+
+It does so by compiling programs and running them, and additionally, checking
+the size of the result binraries.
+"""
+from pathlib import Path
+
+from cijoe.core.resources import get_resources
+
+
+def prep_sources(cijoe):
+    """
+    Transfers sources to target and returns a dict with paths to them
+    """
+
+    resources = get_resources()
+
+    tmp = Path("/tmp")
+    paths = {
+        "loading": tmp / "xnvme_loading.c",
+        "linking": tmp / "xnvme_linking.c",
+    }
+
+    for ltype, path in paths.items():
+        cijoe.put(resources["auxiliary"][f"xnvme_{ltype}"].path, paths[ltype])
+
+    return paths
+
+
+def list_libraries(cijoe):
+    """Returns a list of Paths to xNVMe libraries"""
+
+    suffix = {
+        ".so": "shared",
+        ".dylib": "shared",
+        ".dll": "shared",
+        ".a": "static",
+    }
+
+    err, state = cijoe.run("pkg-config xnvme --variable=libdir")
+    assert not err
+
+    libdir = state.output().strip()
+
+    err, state = cijoe.run(f"ls {libdir}")
+    assert not err
+
+    paths = [
+        Path(libdir) / line
+        for line in state.output().split()
+        if "libxnvme" in Path(line.strip()).stem
+    ]
+
+    return {suffix[lib.suffix]: lib for lib in paths}
+
+
+def test_library_consumption(cijoe):
+    """Verify that the shared library is consumable via dynamic loading"""
```

Comment:
This seems to be inaccurate. It tests all 3 types right?

Reply:
Right! Nice catch. Re-phrased!